src/modify.js

```diff
@@ -66,12 +101,122 @@ function rewriteAllFields(schema, configCallback, context) {
   });
 }
 
+function pickFields(originalSchema, pickConfig) {
+  if (!pickConfig) return originalSchema;
+
+  const fieldsToPick = pickConfig.fields;
+  const newSchema = {
+    properties: {},
+  };
+
+  Object.entries(originalSchema).forEach(([attrKey, attrValue]) => {
+    switch (attrKey) {
+      case 'properties':
+        // TODO — handle recursive nested fieldsets
+        fieldsToPick.forEach((fieldPath) => {
+          set(newSchema.properties, fieldPath, attrValue[fieldPath]);
+        });
+        break;
+      case 'x-jsf-order':
+      case 'required':
+        newSchema[attrKey] = attrValue.filter((fieldName) => fieldsToPick.includes(fieldName));
+        break;
+      case 'allOf': {
+        const newAllOf = [];
+        // remove conditional ("if, then, else") if it does not contain any reference to fieldsToPick
+        originalSchema.allOf.forEach((condition) => {
+          if (isConditionalReferencingAnyPickedField(condition, fieldsToPick)) {
+            newAllOf.push(condition);
+          }
+        });
+        newSchema[attrKey] = newAllOf;
+
+        break;
+      }
+      default:
+        newSchema[attrKey] = attrValue;
+    }
+  });
+
+  // Look for unpicked fields in the conditionals...
+  let missingFields = {};
+
+  newSchema.allOf.forEach((condition, ix) => {
+    const { if: ifCondition, then: thenCondition, else: elseCondition } = condition;
+
+    missingFields = {
+      ...missingFields,
+      ...findMissingFields(ifCondition, {
+        fields: fieldsToPick,
+        path: `allOf[${ix}].if`,
```

Comment:
This `ix` refers to the index of the `newSchema.allOf` and not the `allOf` from the original schema. I think we want to refer to the original index when logging, if that's the case you can use `originalSchema.allOf.indexOf(condition)`.

Reply:
Really nice catch! I modified the test case to catch this one.